---
title: C6318 | Microsoft Docs
ms.date: 11/15/2016
ms.prod: visual-studio-dev14
ms.technology: vs-ide-code-analysis
ms.topic: reference
f1_keywords:
- C6318
helpviewer_keywords:
- C6318
ms.assetid: 3284a83e-bb8e-461c-adcc-cfc66ceea05e
caps.latest.revision: 15
author: corob-msft
ms.author: corob
manager: jillfra
ms.openlocfilehash: 4e590e100ca89e60c59312a2428726cbb01e2f45
ms.sourcegitcommit: 6cfffa72af599a9d667249caaaa411bb28ea69fd
ms.translationtype: MT
ms.contentlocale: ko-KR
ms.lasthandoff: 09/02/2020
ms.locfileid: "77273512"
---
# <a name="c6318"></a>C6318
[!INCLUDE[vs2017banner](../includes/vs2017banner.md)]

경고 C6318: 잘못 정의 된 __try/ \_ _except: 상수 EXCEPTION_CONTINUE_SEARCH 또는 예외 필터 식에서 0으로 계산 되는 다른 상수를 사용 합니다. 예외 처리기 블록의 코드가 실행 되지 않습니다.  
  
 이 경고는이 구조화 된 예외 처리기의 보호 된 블록에서 예외가 발생 한 경우 예외가 `EXCECPTION_CONTINUE_SEARCH` 예외 필터 식에 사용 되기 때문에 예외가 처리 되지 않음을 나타냅니다.  
  
 이 코드는 처리기 블록이 실행 되지 않기 때문에 예외 처리기 블록이 없는 protected 블록에 해당 합니다.  
  
## <a name="example"></a>예  
 다음 코드에서는이 경고를 생성 합니다.  
  
```  
#include <excpt.h>  
#include <stdio.h>  
  
void f (char *pch)  
{  
  __try   
  {  
    // assignment might fail  
    *pch = 0;  
  }   
  __except (EXCEPTION_CONTINUE_SEARCH) // warning 6318  
  {  
    puts("Exception Occurred");  
  }  
}  
```  
  
 이 경고를 해결 하려면 다음 코드를 사용 합니다.  
  
```  
#include <excpt.h>  
#include <stdio.h>  
#include <windows.h>  
  
void f (char *pch)  
{  
  __try   
  {  
    // assignment might fail  
    *pch = 0;  
  }   
  __except( (GetExceptionCode() == EXCEPTION_ACCESS_VIOLATION) ?  
                EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH )  
  {  
    puts("Access violation");  
  }  
}   
```  
  
## <a name="see-also"></a>관련 항목  
 [try-except 문](https://msdn.microsoft.com/library/30d60071-ea49-4bfb-a8e6-7a420de66381)
